T2807 component parameters terms are not ported to po file

Shayan105 opened the pull request:
## Summary 
This PR refactors a lot of code to allow components defined in XML to be translated. 

Linked to PR: https://github.com/CompassionCH/compassion-switzerland/pull/1709

You can see an overview of the .po file diff before and after the changes here : https://www.diffchecker.com/3RgWK5Gr/
**Left : Before the  changes , right:  After the changes**

## Technical explanation 

Lots of components calls receive parameters injection in a way that does not allow translations. 
Namely, these two following patterns disallow translation 

`<t t-set="KEY" t-value=" 'TEXT' " />` 

`<t t-set="KEY" t-value=" 'TEXT1' + some_python_code + 'TEXT2' " />`

Lines following theses type of patterns **inside t-calls** have have been refactored to the following one so that the TEXT are added to the .po file :
`<t t-set="KEY">
      TEXT1 
      <t t-esc="some_python_code" />
      TEXT2
</t>`

#### Note : t-esc changes are only used when the t-value to be changed contains Hard coded string **and**  dynamic values. 

For the following values of KEY : 
- label
- description
- title
- label_empty
- subtitle
- tooltip_title
- placeholder
- button_label
- btn_text
- title_content
- learn_more_text
- text

These changes have been done in the folder my_compassion and the folder my_compassion_switzerland
## Script used (OUTDATED)

The following script has been used to identify lines of codes inside t-call matching the wrong pattern and transforming them into the correct one. 
[script_clean.py](https://github.com/user-attachments/files/23957849/script_clean.py)




### How to use it :
` python3 script_clean.py --force --path <path-module> `

eg: `python3 script_clean.py --force --path my_compassion` will look at the folder named my_compassion and recursively find for .xml file and do the changes. 

the --force arg is for writing the change into disk. 

## Icons, Color and Pictograms handling

To remove a lot of noise in the .po file, the label t-translation= "off"have been added in some places.

Shayan105 committed:
[FIX] [my_compassion] Refactor the componenents format to allow translations

Shayan105 committed:
[FIX][my_compasison] Refactor xml file to allow translations

[FIX] [my_compassion] Refactor the componenents format to allow translations

Revert "[FIX][my_compasison] Refactor xml file to allow translations"

This reverts commit 260aadeeb3210a73a3a23e182fef057dcf035a08.

Shayan105 committed:
[FIX][my_compasison] Refactor xml file to allow translations

Shayan105 committed:
[FIX][my_compassion] Disable translation for icon,colors and pictograms

Shayan105 committed:
[STYLE] Run precommit

Shayan105 committed:
[FIX][my_compassion] Fix syntax and remove useless translation of colors

Shayan105 committed:
[FIX][my_compasiso] Reformat more components value to allow transaltion

Shayan105 committed:
[FIX][my_compassion] Use escaping to make text containing python logic transalatable.

Shayan105 committed:
[FIX][my_compassion] Remove useless text from translation

NoeBerdoz reviewed (changes requested):
Thank you very much for you work! 

There is an issue with the use of t-esc:

Your reasoning of passing dynamic values from t-value into a t-esc is wrong. Those dynamic values are coming from the database, they are not hardcoded string, using t-value is the correct method in that case. We should then on the model level set fields as translatable `translate=True`.

By giving dynamic values as t-esc into the component, you force the value to be treated as string, t-value is better as it keeps the type, so the component instance can then use this variable as its proper type.
 
We should only use your technique for static hardcoded strings or cases where we are sure that the value will be treated as string in all cases.

For cases where there is a mix of dynamic and static values in a t-value, the t-esc is appropriated, as in that case we are sure that we want to treat the value as a string.

Shayan105 committed:
[REVERT] Revert all

This reverts commit 8d6a56f51819740be6e06d492e00ef63512085ae.

Revert "[FIX] [my_compassion] Refactor the componenents format to allow translations"

This reverts commit 3dfbfba80f791c6e3c50bb3232513eae3f4405ca.

Revert "[FIX][my_compasison] Refactor xml file to allow translations"

This reverts commit 56b5896187cbd2cb1230e5e6e019e2da07777496.

Revert "[FIX][my_compassion] Disable translation for icon,colors and pictograms"

This reverts commit 950abe4e8c5abf66d2408c7f757c172f3816edac.

Revert "[STYLE] Run precommit"

This reverts commit 84e7a59aeed6e97928e10a51bf164f27b2c800d3.

Revert "[FIX][my_compassion] Fix syntax and remove useless translation of colors"

This reverts commit ddc9d3f431652ada1b384db2093299b22896aab7.

Revert "[FIX][my_compasiso] Reformat more components value to allow transaltion"

This reverts commit a1ac77ea6b6e3cf5695462af185715ed61d915b6.

Shayan105 committed:
[FIX] [my_compassion] Properly disable translation in colors.xml

Shayan105 committed:
[FIX][My_compassion] Refactor xml to allow translation

Shayan105 committed:
[STYLE] Run precommit

Shayan105 commented:
> Thank you very much for you work!
> 
> There is an issue with the use of t-esc:
> 
> Your reasoning of passing dynamic values from t-value into a t-esc is wrong. Those dynamic values are coming from the database, they are not hardcoded string, using t-value is the correct method in that case. We should then on the model level set fields as translatable `translate=True`.
> 
> By giving dynamic values as t-esc into the component, you force the value to be treated as string, t-value is better as it keeps the type, so the component instance can then use this variable as its proper type.
> 
> We should only use your technique for static hardcoded strings or cases where we are sure that the value will be treated as string in all cases.
> 
> For cases where there is a mix of dynamic and static values in a t-value, the t-esc is appropriated, as in that case we are sure that we want to treat the value as a string.


@NoeBerdoz 
Thank you for your review. The python script and the .xml file have been updated so that the <t-esc> changes apply only if the t-value contains a mix of Hard coded string and dynamic code.
